RoddyCore/src/de/dkfz/roddy/core/ProjectLoader.groovy

```diff
@@ -193,40 +200,54 @@ class ProjectLoader {
     }
 
     void checkAndPreparePremisesOrFail() {
-        if (projectSource == PSRC_UNSET)
-            throw new ProjectLoaderException("There is no project identifier set. It is not possible to run without setting projectSource in the ProjectLoader.")
+        if (projectConfigurationSource == ProjectConfigurationSource.Unset)
+            throw new ProjectLoaderException("There is no project identifier set. It is not possible to run without setting projectConfigurationSource in the ProjectLoader.")
 
-        if (projectSource == PSRC_CONFIGURATION)
+        if (projectConfigurationSource == ProjectConfigurationSource.Configuration)
             return
 
-        if (projectSource == PSRC_PLUGIN) {
+        if (projectConfigurationSource == ProjectConfigurationSource.Automatic) {
             // Create a project configuration, if all premises are valid.
 
             // Check, if the i/o directory is set on the command line
             if (!Roddy.useCustomIODirectories())
                 throw new ProjectLoaderException("It is not possible to use the project configuration free mode without the --${RoddyStartupOptions.useiodir.name()} option.")
 
-            // Create a temporary project configuration in ~/.roddy/configurationFreeMode
-            String configurationFileName = "projects" + projectID + "_" + analysisID + "_" + Integer.toHexString(Roddy.commandLineCall.getOptionValue(RoddyStartupOptions.useiodir).hashCode()) + ".sh"
+            createAndWriteAutoConfigurationFile()
+
+            return
+        }
+
+        throw new ProjectLoaderException("The project identifier is wrong it must either be a configuration or a plugin. Properly set projectConfigurationSource in the ProjectLoader.")
```

Comment:
"... configuration identifier or plugin name ..." ? Would be slightly clearer, or not?